src/chain_sync/network_context.rs

```diff
@@ -363,7 +363,9 @@ where
                     RequestResponseError::ConnectionClosed
                     | RequestResponseError::DialFailure
                     | RequestResponseError::UnsupportedProtocols => {
-                        peer_manager.mark_peer_bad(peer_id).await;
```

Comment:
Hmm this function does not ban peers in `libp2p::Swarm` but remove them from a custom list, we should consider re-writing the logic

Reply:
We have another `PeerManager::ban_peer` which bans peers in `libp2p::Swarm`. While `mark_peer_bad` demotes the chances of peers being selected as network query targets as I understand. I have not read thru the logic but it seems that it removes peers without a duration, we might want to re-write the logic to make it soft-banning at least

```rust
/// Removes a peer from the set and returns true if the value was present
    /// previously
    pub async fn mark_peer_bad(&self, peer_id: PeerId) -> bool {
        let mut peers = self.peers.write().await;
        let removed = remove_peer(&mut peers, &peer_id);
        if removed {
            metrics::FULL_PEERS.dec();
        }

        // Add peer to bad peer set
        debug!("marked peer {} bad", peer_id);
        if peers.bad_peers.insert(peer_id) {
            metrics::BAD_PEERS.inc();
        }

        removed
    }

    /// Remove peer from managed set, does not mark as bad
    pub async fn remove_peer(&self, peer_id: &PeerId) -> bool {
        let mut peers = self.peers.write().await;
        debug!("removed peer {}", peer_id);
        let removed = remove_peer(&mut peers, peer_id);
        if removed {
            metrics::FULL_PEERS.dec();
        }
        removed
    }
```

Reply:
Ah, I see it now, thanks. Shall we temporarily remove the `ban_peer`? If not, what kind of `soft-banning` do you have in mind?

Reply:
Basically, the PeerManager maintains a peer list by itself, to optimize network query performance by using peer latency etc. to prioritize the list. IMHO I don't think it's worth the complexity and I would remove the entire logic, as we have seen we can get big perf improvement by just simply increasing the request window size.  

I'm not sure if the root cause of the issue is that, all peers are banned in libp2p::Swarm or in PeerManager, if this PR fixes the issue, it means PeerManager is buggy and we should maybe just remove it.

Reply:
I'll run it for a while on a droplet. In general, in the issues logs, we don't ban that many peers, but we do mark quite a few as bad.